### ISSUE TITLE
Can bpf_spin_lock directly be used in BCC?

### ISSUE THREAD

**Issue by ganli2015** [iovisor/bcc#4844]
Can I use bpf_spin_lock in BCC like this:

            bpf_spin_lock(&q->lock);  
            u64 stack_id = stack_traces.get_stackid(ctx, STACK_FLAGS);
            bpf_spin_unlock(&q->lock);

**Comment by chenhengqi** [comment]
No, you can't use bpf_spin_lock in tracing progs as of kernel v5.1.
See the conversation [here](https://github.com/iovisor/bcc/pull/4669#discussion_r1392458503).

**Comment by ganli2015** [comment]
> No, you can't use bpf_spin_lock in tracing progs as of kernel v5.1. See the conversation [here](https://github.com/iovisor/bcc/pull/4669#discussion_r1392458503).

And  __sync_lock_test_and_set  or __sync_val_compare_and_swap are also not available yet?

**Comment by chenhengqi** [comment]
This requires `-mcpu=v3` features which is not available on v5.1.
